### PR TITLE
Vk

### PR DESCRIPTION
**Pull request by ValKurta**
test

### CODE REVIEW

**Comment by ValKurta** [comment]
+https
+reload
+fix for refresh_tocken in routes/auth
+CORS
+logout
+black list of tockens
+rate limit via lifespan(on_event("startup") will be decomissed)+redis
+argon to hash_password
+control for http/https mode
+control for hashing
-redundancy

File: README.md [hunk 1]
```diff
@@ -45,3 +45,12 @@ poetry run pre-commit run --all-files
 # commit without pre-commit checks
 git commit -m "commit message" --no-verify
 
+## To generate key.pem (private key) and cert.pem (certificate)
+# Windows(SSL)
+openssl genrsa -out key.pem 2048
+openssl req -new -x509 -key key.pem -out cert.pem -days 365
```
Comment (oleksyii) [inline README.md:51]: maybe refer to how to install openssl on windows in readme

**Review by oleksyii — APPROVED** [review]
Looks good. Maybe add an instruction on how to install openssl to README

File: src/routes/auth.py [hunk 2]
```diff
@@ -82,13 +88,17 @@ async def refresh_token(credentials: HTTPAuthorizationCredentials = Security(oau
...
     """
-    User logout and the addition to the black list of tokens
+    User logout and the addition into the black list of tokens
     """
-    await add_token_to_blacklist(token, db)
-    return {"detail": "Logged out successfully"}
+    try:
+        payload = jwt.decode(token, settings.jwt_secret_key, algorithms=[settings.algorithm])
+        await add_token_to_blacklist(token, db)
+        return {"detail": f"User with token {token} has logged out successfully."}
```
Comment (iamleeon) [inline src/routes/auth.py:99]: Is displaying the token in the "detail" message secure and useful?

**Review by iamleeon — APPROVED** [review]

**Review by iamleeon — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ValKurta) [reply README.md:51]: added prompts what to do in this case according to the comment

File: src/routes/auth.py (hunk 2 above)
Comment (ValKurta) [reply src/routes/auth.py:99]: agree. showed in in the tests and in our demo- moved out from the message

**Pull request merged**